WcaOnRails/app/controllers/polls_controller.rb

```diff
@@ -0,0 +1,81 @@
+class PollsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :can_create_poll_only, only: [:new, :create, :update, :destroy]
+  before_action :can_vote_in_poll_only, only: [:index, :vote, :results]
+
+  def index
+    if current_user.can_create_poll?
+      @polls = Poll.all
+    else
+      @polls = Poll.where(confirmed: true)
+    end
+    @openPolls , @closedPolls = [], []
+    @polls.each do |poll|
+      @openPolls << poll if !poll.poll_is_over?
+      @closedPolls << poll if poll.poll_is_over?
+    end
+  end
+
+  def new
+    @poll = Poll.new
+  end
+
+  def vote
+    @poll = Poll.find(params[:id])
+    @vote = @poll.votes.find_by_user_id(current_user.id) || Vote.new
+  end
+
+  def results
+    @poll = Poll.find(params[:id])
+  end
+
+  def create
+    @poll = Poll.new(poll_params)
+    @poll.multiple = false
+    @poll.deadline = Date.today + 15
+    @poll.confirmed = false
+    if @poll.save
+      flash[:success] = "Created new poll"
+      redirect_to edit_poll_path(@poll)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @poll = Poll.find(params[:id])
+  end
+
+  def update
+    @poll = Poll.find(params[:id])
+    if @poll.update_attributes(poll_params)
+      if params[:commit] == "Confirm"
+        flash[:success] = "Poll confirmed and open to voting"
+        redirect_to polls_path
+      else
+        flash[:success] = "Updated poll"
+        render 'edit'
```

Comment:
It should not. Same thing as with votes, we send them back to editing (to check if everything is right and/or confirm).